lib/templates/android/inc/_attributes_getter_setter.liquid

```diff
@@ -12,28 +13,26 @@
             {%- endif -%}
           {%- endif -%}
         {%- endcapture -%}
-
         {%- capture primitives -%}
           {%- include 'inc/primitives' -%}
         {%- endcapture -%}
-
-        {%- capture annotation -%}
-          {%- if params.support_annotations == true and ((params.use_wrappers == true and attribute.optional == true) or (attribute.enum_type.size > 0 and attribute.optional == true)) -%}
-          @android.support.annotation.Nullable {% endif -%}
-          {%- if params.support_annotations == true and ((attribute.enum_type != primitives and attribute.optional == false) or (attribute.enum_type.size > 0 and attribute.optional == false)) -%}
-          @android.support.annotation.NonNull {% endif -%}
-          {%- if attribute.support_annotation.size > 0 -%}
-            @android.support.annotation.{{ attribute.support_annotation }} {% endif -%}
+        {%- capture annotation -%}{%- if params.support_annotations == true -%}
+        {%- if params.use_wrappers == true or attribute.enum_type.size > 0 or primitives == "false" -%}
+        @android.support.annotation.{% if attribute.optional == true %}Nullable{% else %}NonNull{% endif -%}
+          {%- if attribute.support_annotation.size > 0 %}-{% endif -%}
```

Comment:
Nitpicking: I'd rather use a comma (`,`) than a `-` for this hack, feels less hacky at least, as `,` is a common separator for list items